display_list/display_list_color_filter.h

```diff
@@ -240,6 +247,61 @@ class DlLinearToSrgbGammaColorFilter final : public DlColorFilter {
   friend class DlColorFilter;
 };
 
+class DlComposedColorFilter final : public DlColorFilter {
+ public:
+  DlComposedColorFilter(std::shared_ptr<DlColorFilter> outer,
+                        std::shared_ptr<DlColorFilter> inner)
+      : outer_(std::move(outer)), inner_(std::move(inner)) {}
```

Comment:
Can we make this only have factories that can check outer/inner for null and only return the other?